### PR TITLE
use transitionend to fix safari rendering bug

### PR DESCRIPTION
**Pull request by danyaal**

### CODE REVIEW

File: src/PopEngine/index.ts [hunk 1]
```diff
@@ -146,6 +146,13 @@ export class PopEngine {
         document.body.appendChild(container);
       }
 
+      if (this._isSafari()) {
+        document.querySelector('div[pop-id="' + groupId + '"]').addEventListener('transitionend', function(): void {
+        console.log('transition!!');
```
Comment (matthewHuiCsc) [inline src/PopEngine/index.ts]: Remove console.log :)

File: src/PopEngine/index.ts [hunk 2]
```diff
@@ -146,6 +146,13 @@ export class PopEngine {
         document.body.appendChild(container);
       }
 
+      if (this._isSafari()) {
```
Comment (matthewHuiCsc) [inline src/PopEngine/index.ts]:
```
 +      if (this._isSafari()) {
 +        document.querySelector('div[pop-id="' + groupId + '"]').addEventListener('transitionend', function(): void {
 +        console.log('transition!!');
 +        container.classList.remove('hidden');
 +      }.bind(this), true);
```

When a pop is reused, is another handler created on the same pop? I'm thinking this code needs to remove itself too once transitionend triggers.

File: src/popgun.scss [hunk 3]
```diff
@@ -190,10 +190,12 @@ $boxShadowPop: 1px 2px 2px rgba(#1a333d, 0.15);
   overflow: visible;
   white-space: normal;
 
-  transition: top .25s ease-out, left .25s ease-out;
+  -webkit-transition: top .25s ease-in-out, left .25s ease-in-out;
+  transition: top .25s ease-in-out, left .25s ease-in-out;
```
Comment (matthewHuiCsc) [inline src/popgun.scss]: I don't particularly care that this changed from `ease-out` to `ease-in-out`, but I just wanted to hear the reasoning.

Comment (danyaal) [inline src/popgun.scss]: Uhh that was left over from when I was just playing around with the stylesheet.. good catch! I'll remove it! 😄

File: src/PopEngine/index.ts [hunk 4]
```diff
@@ -146,6 +151,11 @@ export class PopEngine {
         document.body.appendChild(container);
       }
 
+      if (UserAgentUtil.isSafari()) {
```
Comment (matthewHuiCsc) [inline src/PopEngine/index.ts]:
```
 +      if (UserAgentUtil.isSafari()) {
 +        this._transitionendCallbacks[groupId] = this._removeHiddenClass.bind(this);
 +        document.querySelector('div[pop-id="' + groupId + '"]').addEventListener('transitionend', this._transitionendCallbacks[groupId], true);
 +      }
 +
```

I suspect this should be `if(UserAgentUtil.isSafari() && !this._transitionendCallbacks[groupId]) { ... }`
This way we avoid adding the multiple event listeners per pop.

Comment (danyaal) [inline src/PopEngine/index.ts]: oh man nice catch!

**Comment by matthewHuiCsc** [comment]
LGTM! ship all the things!

**Pull request merged**